### PR TITLE
Fix some issues causing aspnetcore-helix-matrix failures

### PR DESCRIPTION
**Pull request by halter73**
Example failures:

https://dev.azure.com/dnceng/public/_build/results?buildId=848189&view=ms.vss-test-web.build-test-results-tab

### CODE REVIEW

**Comment by halter73** [comment]
I manually triggered the aspnetcore-helix-matrix pipeline on the halter73/fix-helix-matrix PR branch and now it's showing up under the PR checks. I wonder if GitHub will block merging now if it fails 🤔

**Review by TanayParikh — APPROVED** [review]

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -43,7 +43,7 @@
   <ItemGroup Condition="'$(TargetArchitecture)' == 'arm64' AND '$(IsHelixDaily)' == 'true' AND '$(IsWindowsOnlyTest)' != 'true'">
     <HelixAvailableTargetQueue Include="(Debian.9.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-a12566d-20190807161036" Platform="Linux" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetArchitecture)' == 'arm64' AND '$(IsHelixDaily)' == 'true'">
+  <ItemGroup Condition="'$(IsWindowsOnlyTest)' != 'true' AND '$(TargetArchitecture)' == 'arm64' AND '$(IsHelixDaily)' == 'true'">
```
Comment (Tratcher) [inline eng/targets/Helix.Common.props:46]: Is there an issue tracking this part? The initial error looked like an easy fix in TestMatrix.

**Review by Tratcher — APPROVED** [review]

File: eng/targets/Helix.Common.props (hunk 1 above)
Comment (TanayParikh) [reply eng/targets/Helix.Common.props:46]: I don't believe so, @halter73 could you please confirm? If not, I'll create a new one.

Comment (halter73) [reply eng/targets/Helix.Common.props:46]: Thanks @TanayParikh, but I got it. https://github.com/dotnet/aspnetcore/issues/26872

**Comment by halter73** [comment]
The recent aspnetcore-helix-matrix failures are related to https://github.com/dotnet/core-eng/issues/10962. I'm merging this since all the PR checks passed.

**Comment by TanayParikh** [comment]
Thanks @halter73!

**Pull request merged**